core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1326,6 +1355,42 @@ int leaderEpoch() {
         return leaderEpoch;
     }
 
+    // Visible for testing
+    void recordFetchLockRatioMetric(long acquiredTime) {
+        // Update the total fetch lock acquired time.
+        double fetchLockToTotalTime;
+        if (acquiredTime + timeSinceLastLockAcquisitionMs == 0) {
+            // If the total time is 0 then the ratio is 1 i.e. the fetch lock was acquired for the complete time.
+            fetchLockToTotalTime = 1;
```

Comment:
nit: `1.0`. We are switching between floating point and integers here so I would be explicit about this. I know it makes no practical difference.

Reply:
Done.